.github/workflows/development-github-pages.yaml

```diff
@@ -0,0 +1,138 @@
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: kroxylicious
+          ref: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.branch || github.ref_name }}
+      
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+  
+      - name: Install asciitosvg
+        run: go install github.com/asciitosvg/asciitosvg/cmd/a2s@latest
+      
+      - name: Build and install BOM
+        run: mvn clean install -pl :kroxylicious-bom
+        working-directory: kroxylicious
+      
+      - name: Build with Maven
+        run: mvn -P dist package --non-recursive
+        working-directory: kroxylicious
+      
+      - name: Checkout Website
+        uses: actions/checkout@v4
+        with:
+          path: kroxylicious.github.io
+          repository: kroxylicious/kroxylicious.github.io
+          ref: main
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: network=host
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: kroxylicious.github.io
+          push: true
+          tags: localhost:5000/kroxy-jekyll:latest
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Load Jekyll configuration overrides
+        run: echo "${{ vars.JEKYLL_CONFIG_OVERRIDES }}" > _config-overrides.yml
+        working-directory: kroxylicious.github.io
+
+      - name: copy built documentation into website
+        run: cp -r kroxylicious/target/web/* kroxylicious.github.io/
+
+      - name: Build with Jekyll
+        run: |
+          docker run \
+            --rm \
+            -u "$(id -u):$(id -g)" \
+            -v "$(pwd):/site" \
+            localhost:5000/kroxy-jekyll:latest \
+            bash -c 'eval "$(rbenv init -)" && cp -r /css/_sass/bootstrap /site/_sass/ && bundle exec jekyll build --config=_config.yml,_config-overrides.yml'
+        env:
+          JEKYLL_ENV: production
+        working-directory: kroxylicious.github.io
+      - name: Get Project Version
+        id: get_version
+        run: echo "VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_OUTPUT
+        working-directory: kroxylicious
+      - name: Upload artifact
+        # Automatically uploads an artifact from the './_site' directory by default
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: kroxylicious.github.io/_site/documentation/${{ steps.get_version.outputs.VERSION }}
```

Comment:
I'm wondering if we are better off deploying the entire website. If we use a particular directory, then it's likely the contained HTML will contain absolute links that expect `${site.url}/${base.url}/documentation/${version}` to exist. 

Maybe we should deploy the entire website and add some banner or warning to the header so you know you are looking at a development version.

Reply:
PR adding a warning banner when the Jekyll environment is `development`: https://github.com/kroxylicious/kroxylicious.github.io/pull/166